Adding the Password Checker package to the composer.json file.

briancolinger opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

Fixes #

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Adds the automattic/jetpack-password-checker package to the composer.json file.

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to the `projects/plugins/jetpack` directory.
* Delete the `vendor` directory and `composer.lock` file.
* `composer update`
*  Verify that the `jetpack-password-checker` package is in the `projects/plugins/jetpack/vendor/automattic` directory.

kbrown9 commented on projects/packages/sync/composer.json:
```diff
@@ -8,6 +8,7 @@
 		"automattic/jetpack-constants": "^1.6",
 		"automattic/jetpack-heartbeat": "^1.3",
 		"automattic/jetpack-options": "^1.11",
+		"automattic/jetpack-password-checker": "0.1.x-dev",
```
I think we want to use '^0.1" for the version here. `0.1.0` is the first release of the `jetpack-password-checker` package.

zinigor replied:
Also I think since we're getting that package as a direct dependency of Sync, we should remove the package from Jetpack's dependencies.

kraftbj replied:
> Also I think since we're getting that package as a direct dependency of Sync, we should remove the package from Jetpack's dependencies.

Assuming that's the only place the Password Checker is being used. Just be sure to confirm that before removing it directly from Jetpack deps.

briancolinger replied:
Fixed.

briancolinger commented:
Ok, this could use a double check for sure.
I think I did it right, but have little confidence. :)

I've removed the password checker package from the Jetpack plugin composer.json and added it to the Sync package composer.json.
I ran `composer update` after these changes and I ended up with changes to the composer.lock file.
Is that normal? Asking for a friend. :)

I also added changelog entries for the removal and addition. I wasn't sure if I should one for each or a single combined entry. I went with adding two different ones to be on the safe side, maybe we can merge them if we only need one.

zinigor commented:
I took the liberty to revert the `composer.lock` changes in the main folder, hope you don't mind.

briancolinger commented:
I don't mind at all, thank you!!

As I was laying in bed trying to go to sleep last night, I happened to think about the .gitignore settings that I may need to check those later on.
That explains the composer.lock mystery for me!!

kbrown9 reviewed (approved):
This looks good to me!

kbrown9 commented:
>It looks like you did all the right things, but the composer update in the main folder was not necessary. I think a composer install in the projects/plugins/jetpack folder should have been enough

I tested by:
1. Going to `projects/plugins/jetpack`.
2. Removing the `vendor` folder and `composer.lock`.
3. Running `composer update` and verifying that the `jetpack-password-checker` package was in Jetpack's `vendor` folder.

briancolinger commented:
Thank you @kbrown9! Your testing instructions are way better than mine!
I've updated the PR's testing instructions.

zinigor reviewed (dismissed):
Thanks, let's get it in!
